### ISSUE TITLE
NullPointerException in Meta.migrateTo() when using DatabaseMetaData backed Meta containing views

### ISSUE THREAD

**Issue by Ubehebe** [jOOQ/jOOQ#10738]
Hi there, thanks for your work on this high-value library. I've been using jooq for a couple years, and this is the first time I've run into a problem that I think is a bug rather than a misconfiguration or user error.

### Expected behavior

I'm trying to learn the new-ish jooq apis for doing schema migrations. I started by writing a simple test: a db migrated to itself should produce an empty migration.

Minimal db to repro:

```sql
CREATE TABLE places (place_name text);
CREATE VIEW new_view AS SELECT place_name FROM places;
```

java test case:

```java
@Test
public void minimalViewRoundTrip() {
  DSLContext minimal = ... // set up from PGSimpleDataSource
  assertThat(minimal.meta().migrateTo(minimal.meta())).isEmpty();
}
```

### Actual behavior

```
1) minimalViewRoundTrip(dbutil.test.JooqSchemaDiffTest)                       
java.lang.NullPointerException                                                
        at org.jooq.impl.Diff$10.merge(Diff.java:273)                         
        at org.jooq.impl.Diff$10.merge(Diff.java:265)                         
        at org.jooq.impl.Diff.append(Diff.java:625)
        at org.jooq.impl.Diff.append(Diff.java:569)
        at org.jooq.impl.Diff.appendTables(Diff.java:315)
        at org.jooq.impl.Diff.access$500(Diff.java:89)
        at org.jooq.impl.Diff$4.merge(Diff.java:167)
        at org.jooq.impl.Diff$4.merge(Diff.java:164)
        at org.jooq.impl.Diff.append(Diff.java:625)
        at org.jooq.impl.Diff.append(Diff.java:569)
        at org.jooq.impl.Diff.appendSchemas(Diff.java:173)
        at org.jooq.impl.Diff.access$000(Diff.java:89)
        at org.jooq.impl.Diff$1.merge(Diff.java:123)
        at org.jooq.impl.Diff$1.merge(Diff.java:120)
        at org.jooq.impl.Diff.append(Diff.java:625)
        at org.jooq.impl.Diff.append(Diff.java:569)
        at org.jooq.impl.Diff.appendCatalogs(Diff.java:112)
        at org.jooq.impl.Diff.queries(Diff.java:108)
        at org.jooq.impl.AbstractMeta.migrateTo(AbstractMeta.java:350)
        at org.jooq.impl.AbstractMeta.migrateTo(AbstractMeta.java:345)
        at dbutil.test.JooqSchemaDiffTest.minimalViewRoundTrip(JooqSchemaDiffTest.java:65)
```

I believe this is the same root cause as #10205: the source code of the view is empty, because it's constructed from jdbc, not directly from ddl. To confirm this, I wrote a second test case that succeeds:

```java
@Test
public void minimalFromDdl() {
  var db = DSL.using(SQLDialect.POSTGRES);
  var meta = db.meta("""
  CREATE TABLE places (place_name text);
  CREATE VIEW new_view AS SELECT place_name FROM places;
  """);
  assertThat(meta.migrateTo(meta)).isEmpty();
}
```

### Versions

- jOOQ: 3.13.5
- Java: 14
- Database (include vendor): postgres 12
- OS: linux and macos (repros on both)
- JDBC Driver (include name if inofficial driver): postgres jdbc 42.2.8

Thanks again for all you do!

**Comment by lukaseder** [comment]
Thanks a lot for your report and for your nice words. I can reproduce this. Will look into it right away.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.14.0. Backport scheduled for 3.13.6 (#10740). Thanks again for your report!

**Comment by lukaseder** [comment]
A related issue has been discovered in the integration tests. Might not affect you as it is SQLite specific: https://github.com/jOOQ/jOOQ/issues/10741
